### PR TITLE
[FS] Dynamic Fees

### PR DESCRIPTION
**Pull request by quantumagi**

### CODE REVIEW

File: src/Stratis.Features.FederatedPeg/FederationGatewaySettings.cs [hunk 1]
```diff
@@ -113,7 +112,10 @@ public FederationGatewaySettings(NodeSettings nodeSettings, FederatedPegOptions
         public int MultiSigN { get; }
 
         /// <inheritdoc/>
-        public Money TransactionFee { get; }
+        public Money TransactionFee(int numInputs)
+        {
+            return DefaultTransactionFee + numInputs * InputsTransactionFee;
```
Comment (dangershony) [inline src/Stratis.Features.FederatedPeg/FederationGatewaySettings.cs]: Currently `Money.Coins(0.0m)` I this correct?

File: src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs [hunk 2]
```diff
@@ -434,13 +434,15 @@ public Task<bool> RecordLatestMatureDepositsAsync(IList<MaturedBlockDepositsMode
                                 CrossChainTransferStatus status = CrossChainTransferStatus.Suspended;
                                 Script scriptPubKey = BitcoinAddress.Create(deposit.TargetAddress, this.network).ScriptPubKey;
 
-                                if (!haveSuspendedTransfers)
+                            if (!haveSuspendedTransfers)
+                            {
+                                var fixedTransferCost = new Money(0.01m, MoneyUnit.BTC);
```
Comment (dangershony) [inline src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs]: is this some randmon number? it should be a const

File: src/Stratis.Features.FederatedPeg/FederationGatewaySettings.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg/FederationGatewaySettings.cs]: Yes, Until we decide to make this something else.

File: src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs (hunk 2 above)
Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs]: It will be. Was focusing on the concept and first.

**Comment by codingupastorm** [comment]
@quantumagi whilst I hear you're away I'll close this and open a new PR that I'm able to edit

**Pull request closed without merging**